pkg/common/support/collconfigretriever.go

```diff
@@ -16,41 +16,65 @@ import (
 	"github.com/hyperledger/fabric/core/common/ccprovider"
 	"github.com/hyperledger/fabric/core/common/privdata"
 	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/extensions/collections/api/support"
+	"github.com/hyperledger/fabric/extensions/endorser/api"
 	gossipapi "github.com/hyperledger/fabric/extensions/gossip/api"
-	mspmgmt "github.com/hyperledger/fabric/msp/mgmt"
+	"github.com/hyperledger/fabric/msp"
 	"github.com/hyperledger/fabric/protos/common"
 	"github.com/pkg/errors"
+	collcommon "github.com/trustbloc/fabric-peer-ext/pkg/collections/common"
 )
 
-type collectionConfigRetrieverCache struct {
-	mutex      sync.RWMutex
-	retrievers map[string]*CollectionConfigRetriever
+type ledgerProvider interface {
+	GetLedger(cid string) ledger.PeerLedger
 }
 
-var collConfigRetrieverCache = &collectionConfigRetrieverCache{
-	retrievers: make(map[string]*CollectionConfigRetriever),
+// CollectionConfigRetrieverProvider is a collection config retriever provider
+type CollectionConfigRetrieverProvider struct {
+	mutex                  sync.RWMutex
+	retrievers             map[string]*CollectionConfigRetriever
+	ledgerProvider         ledgerProvider
+	blockPublisherProvider api.BlockPublisherProvider
+	idProvider             collcommon.IdentityDeserializerProvider
 }
 
-// CollectionConfigRetrieverForChannel returns the collection config retriever for the given channel
-func CollectionConfigRetrieverForChannel(channelID string) *CollectionConfigRetriever {
-	return collConfigRetrieverCache.forChannel(channelID)
+// NewCollectionConfigRetrieverProvider returns a new CollectionConfigRetrieverProvider
+func NewCollectionConfigRetrieverProvider(ledgerProvider ledgerProvider, blockPublisherProvider api.BlockPublisherProvider, idProvider collcommon.IdentityDeserializerProvider) *CollectionConfigRetrieverProvider {
+	logger.Info("Creating collection config retriever provider")
```

Comment:
is this logging necessary? if yes, would it help ops to log `idProvider` maybe?

Reply:
This is informative in that the resource was properly started. If another other resource depended on this one and could not find it then I would be looking for this log and the order in which the resources were started. ID provider has no useful info to log.